code/particle/effects/GenericShapeEffect.h

```diff
@@ -156,6 +165,26 @@ class GenericShapeEffect : public ParticleEffect {
 		if (internal::required_string_if_new("+Number:", nocreate)) {
 			m_particleNum = ::util::parseUniformRange<uint>();
 		}
+		if (!nocreate) {
+			m_particleChance = 1.0f;
+		}
+		if (optional_string("+Chance:")) {
+			float chance;
+			stuff_float(&chance);
+			CLAMP(chance, 0.0f, 1.0f);
+			if (chance <= 0.0f) {
+				Error(LOCATION,
```

Comment:
You've already clamped `chance`, so it can't be less than zero. Additionally, this doesn't seem like an unrecoverable problem, so I'm not sure why this is an `Error` while the similar `> 1.0f` case (which, again, can't execute because the value has already been clamped) is only a `Warning`.

Reply:
the point about clamping is true. I must have written that before deciding to do the more elaborate handling and forgotten to remove it.
My thinking with warning vs error was rooted in just a gut sense of how wrong each case was, a >100% chance to spawn is arguably modeled well by 100%, but I have no idea what someone who set for <0% wants. But, it would make sense to treat both as warnings now that you question it.

Reply:
Well, regardless of what they *wanted*, they've clearly told the engine never to spawn the particle. Which gives a very simple way to recover from this issue: never spawn the particle. Therefore it doesn't really make sense (to me) to halt program execution here.

Reply:
Fair, I'll adjust.